### PR TITLE
Dropdown Fix RightAligned

### LINKED ISSUE #3682: RightAligned of dropdown not work well
I am using Blazorise.Bootstrap5
Code:
```razor
<Dropdown RightAligned="true">
    <DropdownToggle Color="Color.Primary">
        Dropdown
    </DropdownToggle>
    <DropdownMenu>
        <DropdownItem>Action</DropdownItem>
        <DropdownDivider />
        <DropdownItem>Another Action</DropdownItem>
    </DropdownMenu>
</Dropdown>
```
result( it is not right aligned):
image

project:
[BlazorServerApp.zip](https://github.com/Megabit/Blazorise/files/8423122/BlazorServerApp.zip)

### PR DESCRIPTION
**Pull request by David-Moreira**
Closes #3682

FIxes RightAligned parameter no longer being considered, since we've introduced popper to handle dropdown menu positioning.
And like we've talked also sets up better popper overflow/out of bounds detection, moving the menu accordingly.

### CODE REVIEW

**Comment by stsrki** [comment]
Tested and alignment working well so far.

There is one funny issue. If you click anywhere inside of marked area the menu will not close 😂

image

**Comment by stsrki** [comment]
BS5, Bulma, and AntDesign are not working.

File: Source/Blazorise/wwwroot/dropdown.js [hunk 1]
```diff
@@ -44,19 +47,23 @@ export function initialize(element, elementId, targetElementId, altTargetElement
...
                     padding: 0,
                 }
-            }]
+            }
+        ]
     });
 
     instance.update();
 
+    createAttributesObserver(menuElement.id, (mutationsList) => observeClassChanged(mutationsList, options.dropdownShowClassName, () => instance.update()));
```
Comment (stsrki) [inline Source/Blazorise/wwwroot/dropdown.js:65]: What if `options.dropdownShowClassName` is null?

Comment (David-Moreira) [reply Source/Blazorise/wwwroot/dropdown.js]: then it certainly looks for a null class which will never find I guess. I'll add a null check with an early return in the observer

**Comment by David-Moreira** [comment]
> BS5, Bulma, and AntDesign are not working.

It should now be working, I hadn't time to test/adequate at the time.

**Comment by David-Moreira** [comment]
> > BS5, Bulma, and AntDesign are not working.
> 
> It should now be working, I hadn't time to test/adequate at the time.

I didnt do anything in BS5, I was just re-testing and for a moment it seemed like it was not working since I forgot to clear the cache. Can you check if you've cleared and try again?

**Comment by stsrki** [comment]
All is working now except for AntDesign. When you open menu first time it works, but as soon as you close and reopen it it show like:

image

File: Source/Blazorise/Components/Dropdown/Dropdown.razor.cs [hunk 2]
```diff
@@ -70,11 +70,14 @@ protected override void OnAfterRender( bool firstRender )
...
                     altTargetElementId: childrenButtonList?.FirstOrDefault()?.ElementId,
                     menuElementId: childrenDropdownMenus?.FirstOrDefault()?.ElementId,
+                    showElementId: GetShowElementId(),
                     options: new
                     {
                         Direction = GetDropdownDirection().ToString( "g" ),
+                        RightAligned = RightAligned,
                         DropdownToggleClassNames = ClassProvider.DropdownToggle( IsDropdownSubmenu ),
                         DropdownMenuClassNames = ClassProvider.DropdownMenu(),
+                        DropdownShowClassName = GetShowClassName()
```
Comment (stsrki) [inline Source/Blazorise/Components/Dropdown/Dropdown.razor.cs:80]: Why is this not part of the ClassProvider? And if it behaves differently for some providers then make use of the BehaviorProvider.

File: Source/Blazorise/Interfaces/Modules/IJSDropdownModule.cs [hunk 3]
```diff
@@ -19,9 +19,10 @@ public interface IJSDropdownModule : IBaseJSModule,
         /// <param name="targetElementId">Target element for the dropdown menu.</param>
         /// <param name="altTargetElementId">An alternative target element for the dropdown menu, usually for split dropdowns.</param>
         /// <param name="menuElementId">Dropdown menu element id.</param>
+        /// <param name="showElementId">Id for the target element that will be listening to the 'show event'.</param>
         /// <param name="options">Additional options for the button initialization.</param>
         /// <returns>A task that represents the asynchronous operation.</returns>
-        ValueTask Initialize( ElementReference elementRef, string elementId, string targetElementId, string altTargetElementId, string menuElementId, object options );
+        ValueTask Initialize( ElementReference elementRef, string elementId, string targetElementId, string altTargetElementId, string menuElementId, string showElementId, object options );
```
Comment (stsrki) [inline Source/Blazorise/Interfaces/Modules/IJSDropdownModule.cs:25]: Seems to me that `showElementId` is the same as `menuElementId`.

Comment (stsrki) [reply Source/Blazorise/Interfaces/Modules/IJSDropdownModule.cs:25]: Hmm maybe not...

Comment (David-Moreira) [reply Source/Blazorise/Interfaces/Modules/IJSDropdownModule.cs:25]: Yes, Bulma has to override.

File: Source/Blazorise/Components/Dropdown/Dropdown.razor.cs (hunk 2 above)
Comment (David-Moreira) [reply Source/Blazorise/Components/Dropdown/Dropdown.razor.cs:80]: Right, I guess I didn't use ClassProvider, cuz it's based on an existing Class Definition. But I'll makle the change.

File: Source/Blazorise/Interfaces/Modules/IJSDropdownModule.cs (hunk 3 above)
Comment (David-Moreira) [reply Source/Blazorise/Interfaces/Modules/IJSDropdownModule.cs:25]: > All is working now except for AntDesign. When you open menu first time it works, but as soon as you close and reopen it it show like:
> 
> image

I think I broke it on merge or smthg, fixing it....

**Review by stsrki — APPROVED** [review]
Tested and it's working well now. AntDesign is also working even better than before. Good job!

**Pull request merged**